### DISCUSSION TITLE
Prisma bundle is too large for Cloudflare Workers

### DISCUSSION THREAD

**Discussion by OpenSource03** [prisma/prisma#24806] [Q&A] [answered]
### Question

Prisma by default is very large.

A way I could previously make it smaller was by using --no-engine, which works great, but unfortunately, doesn't work for the edge environment, as when using adapters, Prisma requires the engine to be included.

I created a very simple app with Clerk (you can also try NextAuth) and Prisma via next-on-(cloudflare)pages.

There are only a couple of backend endpoints, however, it seems like if you have any other moderately larger library with Prisma, you're doomed. Prisma alone works, but including something like NextAuth or Clerk which takes 200-300Kbit makes it too large for Workers free plan resulting into

`Error: Failed to publish your Function. Got error: Your Functions script is over the 1 MiB size limit (workers.api.error.script_too_large)`

  00:53:57.673 | ┌────────────────────────────────────────────────────────────────────────────────┬───────────────┬─────────────┐
  -- | --
  00:53:57.674 | │ Name                                                                           │ Type          │ Size        │
  00:53:57.674 | ├────────────────────────────────────────────────────────────────────────────────┼───────────────┼─────────────┤
  00:53:57.674 | │ __next-on-pages-dist__/cache/adaptor.js                                        │ esm           │ 2.29 KiB    │
  00:53:57.674 | ├────────────────────────────────────────────────────────────────────────────────┼───────────────┼─────────────┤
  00:53:57.674 | │ __next-on-pages-dist__/cache/cache-api.js                                      │ esm           │ 0.43 KiB    │
  00:53:57.674 | ├────────────────────────────────────────────────────────────────────────────────┼───────────────┼─────────────┤
  00:53:57.675 | │ __next-on-pages-dist__/cache/kv.js                                             │ esm           │ 0.35 KiB    │
  00:53:57.675 | ├────────────────────────────────────────────────────────────────────────────────┼───────────────┼─────────────┤
  00:53:57.675 | │ __next-on-pages-dist__/functions/api/v1/info.func.js                           │ esm           │ 10.11 KiB   │
  00:53:57.675 | ├────────────────────────────────────────────────────────────────────────────────┼───────────────┼─────────────┤
  00:53:57.675 | │ __next-on-pages-dist__/functions/api/v1/posts/upvote.func.js                   │ esm           │ 8.45 KiB    │
  00:53:57.675 | ├────────────────────────────────────────────────────────────────────────────────┼───────────────┼─────────────┤
  00:53:57.676 | │ __next-on-pages-dist__/functions/api/v1/posts.func.js                          │ esm           │ 97.34 KiB   │
  00:53:57.676 | ├────────────────────────────────────────────────────────────────────────────────┼───────────────┼─────────────┤
  00:53:57.676 | │ __next-on-pages-dist__/functions/auth/signin/[[...sign-in]].func.js            │ esm           │ 27.45 KiB   │
  00:53:57.676 | ├────────────────────────────────────────────────────────────────────────────────┼───────────────┼─────────────┤
  00:53:57.676 | │ __next-on-pages-dist__/functions/auth/signup/[[...sign-up]].func.js            │ esm           │ 27.37 KiB   │
  00:53:57.676 | ├────────────────────────────────────────────────────────────────────────────────┼───────────────┼─────────────┤
  00:53:57.676 | │ __next-on-pages-dist__/functions/middleware.func.js                            │ esm           │ 174.39 KiB  │
  00:53:57.676 | ├────────────────────────────────────────────────────────────────────────────────┼───────────────┼─────────────┤
  00:53:57.677 | │ __next-on-pages-dist__/manifest/27d10e462fba646daaa29468a0656e12.js            │ esm           │ 1.04 KiB    │
  00:53:57.677 | ├────────────────────────────────────────────────────────────────────────────────┼───────────────┼─────────────┤
  00:53:57.677 | │ __next-on-pages-dist__/manifest/__RSC_SERVER_MANIFEST.js                       │ esm           │ 0.59 KiB    │
  00:53:57.677 | ├────────────────────────────────────────────────────────────────────────────────┼───────────────┼─────────────┤
  00:53:57.677 | │ __next-on-pages-dist__/wasm/wasm_c8adfcbd388903f30d180b0cf275456ec8dd3540.wasm │ compiled-wasm │ 2092.15 KiB │
  00:53:57.677 | ├────────────────────────────────────────────────────────────────────────────────┼───────────────┼─────────────┤
  00:53:57.677 | │ __next-on-pages-dist__/webpack/08d2e599fdb3bd0702cf42d2ab9e89da.js             │ esm           │ 78.74 KiB   │
  00:53:57.677 | ├────────────────────────────────────────────────────────────────────────────────┼───────────────┼─────────────┤
  00:53:57.677 | │ __next-on-pages-dist__/webpack/1054.js                                         │ esm           │ 0.04 KiB    │
  00:53:57.678 | ├────────────────────────────────────────────────────────────────────────────────┼───────────────┼─────────────┤
  00:53:57.678 | │ __next-on-pages-dist__/webpack/65ac3476f933913583f4cf60b0416c7a.js             │ esm           │ 306.89 KiB  │
  00:53:57.678 | ├────────────────────────────────────────────────────────────────────────────────┼───────────────┼─────────────┤
  00:53:57.678 | │ __next-on-pages-dist__/webpack/9a2112b0893561dfed3b0192548c0e47.js             │ esm           │ 91.62 KiB   │
  00:53:57.679 | ├────────────────────────────────────────────────────────────────────────────────┼───────────────┼─────────────┤
  00:53:57.680 | │ __next-on-pages-dist__/webpack/ddf84e86285b7432c6d30c3871b85264.js             │ esm           │ 1.23 KiB    │
  00:53:57.681 | ├────────────────────────────────────────────────────────────────────────────────┼───────────────┼─────────────┤
  00:53:57.681 | │ __next-on-pages-dist__/webpack/ff9105809f416c450bfd7f99269fc4f1.js             │ esm           │ 674.72 KiB  │
  00:53:57.681 | └────────────────────────────────────────────────────────────────────────────────┴───────────────┴─────────────┘

Without engine

  
  00:22:19.675 | ┌─────────────────────────────────────────────────────────────────────┬──────┬────────────┐
  -- | --
  00:22:19.675 | │ Name                                                                │ Type │ Size       │
  00:22:19.676 | ├─────────────────────────────────────────────────────────────────────┼──────┼────────────┤
  00:22:19.676 | │ __next-on-pages-dist__/cache/adaptor.js                             │ esm  │ 2.29 KiB   │
  00:22:19.676 | ├─────────────────────────────────────────────────────────────────────┼──────┼────────────┤
  00:22:19.676 | │ __next-on-pages-dist__/cache/cache-api.js                           │ esm  │ 0.43 KiB   │
  00:22:19.676 | ├─────────────────────────────────────────────────────────────────────┼──────┼────────────┤
  00:22:19.676 | │ __next-on-pages-dist__/cache/kv.js                                  │ esm  │ 0.35 KiB   │
  00:22:19.676 | ├─────────────────────────────────────────────────────────────────────┼──────┼────────────┤
  00:22:19.676 | │ __next-on-pages-dist__/functions/api/v1/info.func.js                │ esm  │ 9.69 KiB   │
  00:22:19.676 | ├─────────────────────────────────────────────────────────────────────┼──────┼────────────┤
  00:22:19.677 | │ __next-on-pages-dist__/functions/api/v1/posts/upvote.func.js        │ esm  │ 8.20 KiB   │
  00:22:19.677 | ├─────────────────────────────────────────────────────────────────────┼──────┼────────────┤
  00:22:19.677 | │ __next-on-pages-dist__/functions/api/v1/posts.func.js               │ esm  │ 96.84 KiB  │
  00:22:19.677 | ├─────────────────────────────────────────────────────────────────────┼──────┼────────────┤
  00:22:19.677 | │ __next-on-pages-dist__/functions/auth/signin/[[...sign-in]].func.js │ esm  │ 27.21 KiB  │
  00:22:19.677 | ├─────────────────────────────────────────────────────────────────────┼──────┼────────────┤
  00:22:19.677 | │ __next-on-pages-dist__/functions/auth/signup/[[...sign-up]].func.js │ esm  │ 27.13 KiB  │
  00:22:19.677 | ├─────────────────────────────────────────────────────────────────────┼──────┼────────────┤
  00:22:19.677 | │ __next-on-pages-dist__/manifest/27d10e462fba646daaa29468a0656e12.js │ esm  │ 1.04 KiB   │
  00:22:19.677 | ├─────────────────────────────────────────────────────────────────────┼──────┼────────────┤
  00:22:19.677 | │ __next-on-pages-dist__/manifest/__RSC_SERVER_MANIFEST.js            │ esm  │ 0.59 KiB   │
  00:22:19.677 | ├─────────────────────────────────────────────────────────────────────┼──────┼────────────┤
  00:22:19.677 | │ __next-on-pages-dist__/webpack/01cf91e6d88885c1a1b1f2f016f92891.js  │ esm  │ 293.20 KiB │
  00:22:19.677 | ├─────────────────────────────────────────────────────────────────────┼──────┼────────────┤
  00:22:19.677 | │ __next-on-pages-dist__/webpack/07e31be319dfea3dadd147e5873dc00d.js  │ esm  │ 674.72 KiB │
  00:22:19.677 | ├─────────────────────────────────────────────────────────────────────┼──────┼────────────┤
  00:22:19.678 | │ __next-on-pages-dist__/webpack/9a2112b0893561dfed3b0192548c0e47.js  │ esm  │ 91.62 KiB  │
  00:22:19.678 | ├─────────────────────────────────────────────────────────────────────┼──────┼────────────┤
  00:22:19.678 | │ __next-on-pages-dist__/webpack/9b0046ce6336b3ded98b802524e830b8.js  │ esm  │ 79.77 KiB  │
  00:22:19.678 | └─────────────────────────────────────────────────────────────────────┴──────┴────────────┘

I believe the largest problem is the compiled-wasm of 2Mbit+ that is included in the output (which is probably the gigantic Prisma engine)...

Are there any ways to make Prisma take less space?

### How to reproduce (optional)

Make a simple page with Prisma and Clerk or NextAuth via https://github.com/cloudflare/next-on-pages

### Expected behavior (optional)

A desired behavior would be to make it possible to use --no-engine in the cloud with adapters feature.

**Comment by ludralph** [comment, marked as answer]
Hi @OpenSource03 👋 

Thank you for raising this question.

You can find a relevant discussion addressing this issue [here](https://github.com/prisma/prisma/discussions/24087#discussioncomment-9339222).

**Reply by OpenSource03** [reply]
Hi @ludralph.

Thanks for pointing out to that discussion.

So basically, Prisma still can not be used on CloudFlare workers... Any ideas on possible future improvements on this? :(
